src/pcre2_jit_compile.c

```diff
@@ -7706,7 +7706,10 @@ static void compile_xclass_matchingpath(compiler_common *common, PCRE2_SPTR cc,
 DEFINE_COMPILER;
 jump_list *found = NULL;
 jump_list **list = (cc[0] & XCL_NOT) == 0 ? &found : backtracks;
-sljit_uw c, charoffset, max = 256, min = READ_CHAR_MAX;
+/* These values should never be negative, suggesting the use of sljit_u32.
+However, they should also remain (far) below 1 << 31, and (would-be)
+negative *differences* won't otherwise undergo correct sign extension. */
+sljit_s32 c, charoffset, max = 256, min = READ_CHAR_MAX;
```

Comment:
c can be UINT32_MAX, so this is also a bad change.

Reply:
Good point.  I've scrapped all changes to `compile_xclass_matchingpath` in favor of formally making `read_char`'s `min` and `max` `sljit_uw`.